Civi/Api4/Query/Api4SelectQuery.php

```diff
@@ -192,11 +206,15 @@ protected function buildWhereClause() {
    * @inheritDoc
    */
   protected function buildOrderBy() {
-    foreach ($this->orderBy as $fieldName => $dir) {
+    foreach ($this->orderBy as $item => $dir) {
       if ($dir !== 'ASC' && $dir !== 'DESC') {
-        throw new \API_Exception("Invalid sort direction. Cannot order by $fieldName $dir");
+        throw new \API_Exception("Invalid sort direction. Cannot order by $item $dir");
```

Comment:
Are item & dir safely escaped by the time they get here?

Reply:
Err, does a variable have to be escaped before passing it into an Exception?

Reply:
Yep, wouldn't be the right layer for escaping/encoding. We've had weird things like [CIVI-SA-2018-03](https://civicrm.org/advisory/civi-sa-2018-03-reflected-xss-in-error-message) in the past where errors that bubble up to the UI may not be escaped, but that's the responsibility of the template (which is why we now purify at that layer).